plugins/hls-code-range-plugin/src/Ide/Plugin/CodeRange.hs

```diff
@@ -126,6 +148,42 @@ findPosition pos root = go Nothing root
             startOfRight <- _start . _codeRange_range <$> V.headM right
             if pos < startOfRight then binarySearchPos left else binarySearchPos right
 
+findFoldingRanges :: CodeRange -> [Maybe FoldingRange]
```

Comment:
These functions share a lot of code with the functions for creating selection ranges. I wonder if we can share some of it.

Reply:
I think both of them now work differently. Can something be done now to share code between the two?

Reply:
Yeah, now they're completely different!